core/storage/trie/polkadot_trie/polkadot_trie_cursor.hpp

```diff
@@ -53,6 +57,13 @@ namespace kagome::storage::trie {
     // in the top entry is the same as \param parent
     void updateLastVisitedChild(const BranchPtr &parent, uint8_t child_idx);
 
+    /**
+     * Constructs a list of branch nodes on the path from the root to the node
+     * with the given \arg key
+     */
+    auto constructLastVisitedChildPath(const common::Buffer &key)
+        -> outcome::result<std::list<std::pair<BranchPtr, int8_t>>>;
```

Comment:
maybe better make a structure instead of pair? return value looks monstrous..

Reply:
Agreed, will look nicer.